latex_directive_completions.py

```diff
@@ -0,0 +1,222 @@
+
+    locales.extend(installed_locales)
+
+    def get_locale(loc):
+        try:
+            loc = detect_spellcheck.normalize_locale(loc)
+            dic = user_sc.get(loc) or detect_spellcheck.get_dict_path(loc)
+            if ac:
+                _, dic = os.path.split(dic)
+            elif dic.startswith("Packages/"):
+                dic = dic[len("Packages/"):]
+        except:
+            dic = "locale"
+        return dic
+    locales = [
+        loc
+        for loc in map(_prettify_locale, locales)
+        if loc.startswith(value)
+    ]
+
+    if ac:
+        comp = [
+            ("{0}\t{1}".format(loc, get_locale(loc)), loc)
+            for loc in locales
+        ]
+    else:
+        comp = [[loc, get_locale(loc)] for loc in locales], locales
+    return comp
+
+
+def _directive_program_completions(view, value, ac=True):
+    engines = [
+        "pdflatex", "xelatex", "lualatex", "pdftex", "xetex", "luatex"
+    ]
+    engines = [e for e in engines if e.startswith(value)]
+    if ac:
+        comp = [(e + "\ttex-program", e) for e in engines]
+    else:
+        comp = engines, engines
+    return comp
+
+
+_LINE_RE = re.compile(
+    r"\s*%\s*!TEX\s+"
+    r"(?P<directive>[\w-]+)(?P<spaces>\s*)"
+    r"=(?P<postspaces>\s*)"
+    r"(?P<prefix>.*)"
+)
+
+_REVERSE_LINE_MATCH_RE = re.compile(
+    r".*\s*=\s*(\w+)\s+XET!\s*%\s*$"
+)
+
+
+class DirectiveFillAllHelper(FillAllHelper):
+
+    def _get_completions(self, view, prefix, line, ac=False):
+        m = re.match(_LINE_RE, line)
+        if not m:
+            return []
+        directive = m.group("directive").lower()
+        # remove leading TS-
+        if directive.startswith("ts-"):
+            directive = directive[3:]
+
+        value = m.group("prefix")
+
+        function = "_directive_{0}_completions".format(directive)
+        # call the completion
+        try:
+            comp = globals().get(function)(view, value, ac)
+        except:
+            comp = []
+
+        if not ac and not prefix and m.group("spaces"):
+            comp = comp[0], [" " + c for c in comp[1]]
+
+        return comp
+
+    def get_auto_completions(self, view, prefix, line):
+        comp = self._get_completions(view, prefix, line, ac=True)
+        return comp
+
+    def get_completions(self, view, prefix, line):
+        comp = self._get_completions(view, prefix, line, ac=False)
+        return comp
+
+    def matches_line(self, line):
+        return bool(_REVERSE_LINE_MATCH_RE.match(line))
+
+    def get_supported_scope_selector(self):
+        return "comment"
+
+    def is_enabled(self):
+        return get_setting("tex_directive_auto_trigger", True)
+
+
+class LatexDirectiveCompletion(sublime_plugin.EventListener):
+    def on_query_completions(self, view, prefix, locations):
+        if len(locations) > 1:
+            return
+        point = locations[0]
+        if not view.score_selector(
+                point, "text.tex.latex comment.line.percentage.tex"):
```

Comment:
Unlike the other uses of the `comment.line.percentage.tex` this actually does the right thing on ST2.